lib/DxcSupport/FileIOHelper.cpp

```diff
@@ -612,6 +612,15 @@ HRESULT DxcCreateBlob(
 
   // Handle empty blob
   if (emptyString) {
+    // Free the passed in pointer if we are supposed to be taking
+    // ownership of it. It is a bit strange to have allocated a zero-sized
+    // buffer, but the IMalloc docs explicitly allow it:
+    //
+    // "If cb is zero, Alloc allocates a zero-length item and returns a valid pointer to that item"
```

Comment:
This case can happen with non-zero sized buffer when it's just a null-terminator, or BOM + null-terminator (but BOM in this scenario has another bug that would blow this up, which needs fixing).